python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -238,7 +238,8 @@ def keep_node(node_id: NodeID) -> None:
                 node_type = tags[TAG_RAY_USER_NODE_TYPE]
                 node_type_counts[node_type] += 1
 
-        def schedule_node_termination(node_id: NodeID, reason: str) -> None:
+        def schedule_node_termination(node_id: NodeID,
+                                      reason: Optional[str]) -> None:
```

Comment:
In the current use case reason  has to always be != None, so I am in favor of keeping this str.

Reply:
I do like bing explicit and returning None instead of empty string.
Adding a check inside schedule_node_termination - this way it is clear reason is non-None.